### PR TITLE
Beminfo 437

### PR DESCRIPTION
**Pull request by tormozz48**
Implement supporting of all existed block documentation and example formats on bem-site-engine level.

@tadatuta @tavriaforever

### CODE REVIEW

File: src/blocks/common.blocks/block-docs/block-docs.bemtree [hunk 1]
```diff
@@ -1,31 +1,71 @@
-block block-docs {
-
-    content: {
-
-        var _data = this.data.node.source.data;
+block('block-docs')(
+    content()(function() {
+        var _data = this.data.node.source.data,
```
Comment (tadatuta) [inline src/blocks/common.blocks/block-docs/block-docs.bemtree]: put `this.data` and `_data[lang]` into variables

File: src/blocks/common.blocks/block-docs/block-docs.bemtree [hunk 2]
```diff
@@ -1,31 +1,71 @@
...
+                    ctx: {
+                        examples: [{ url: p1 }],
+                        source: this.data.node.source
+                    }
+                } ));
+                return sign;
+            });
+
+            description.split(sign).forEach(function(item, index) {
+                buf.push(item);
```
Comment (tadatuta) [inline src/blocks/common.blocks/block-docs/block-docs.bemtree]: buf.push(item, inlineExamples[index]);

File: src/blocks/common.blocks/block-docs/block-docs.bemtree [hunk 3]
```diff
@@ -1,31 +1,71 @@
...
+
+    mode('description')(
+        (function() {
+            var buf = [],
+                sign = '<!--bem-example-->',
+                inlineExamples = [],
+                description = this.ctx.description;
+
+            //parse inline examples
+            /*
```
Comment (tadatuta) [inline src/blocks/common.blocks/block-docs/block-docs.bemtree:24]: if this part is commented, do we really need to keep it here?

File: src/blocks/common.blocks/block-docs/block-docs.bemtree [hunk 4]
```diff
@@ -44,27 +84,31 @@ block block-docs {
...
+            ];
+        }),
 
-    mod-values: {
-
-        if(this.ctx.values && this.ctx.values.length > 0) {
+        match(function() {
+            return !this.ctx.mods || !this.ctx.mods.length;
+        })(function() {
+            return  false;
```
Comment (tadatuta) [inline src/blocks/common.blocks/block-docs/block-docs.bemtree]: extra space.
do we really need `false` here?

File: src/blocks/common.blocks/block-docs/block-docs.bemtree [hunk 5]
```diff
@@ -44,27 +84,31 @@ block block-docs {
...
+        })
+    ),
 
+    mode('mods-title')(function() {
+        return {
+            elem: 'title',
+            tag: 'h4',
```
Comment (tadatuta) [inline src/blocks/common.blocks/block-docs/block-docs.bemtree]: why do we set `tag` and `mix` in BEMTREE?

File: src/blocks/common.blocks/block-docs/block-docs.bemtree (hunk 3 above)
Comment (tormozz48) [inline src/blocks/common.blocks/block-docs/block-docs.bemtree:24]: Yes it is suitable for future development

**Pull request merged**